webpack/assets/javascripts/foreman_toast_notifications.js

```diff
@@ -0,0 +1,40 @@
+import * as ToastActions from './react_app/redux/actions/toasts';
+import store from '../javascripts/react_app/redux';
+import $ from 'jquery';
+/* eslint-disable max-params  */
+
+export function notify(message, type, link, sticky) {
+  showToast({
+    message,
+    type,
+    sticky,
+    link
+  });
+}
+
+function showToast(toast) {
+  store.dispatch(ToastActions.addToast(toast));
+}
+
+// to accommodate rails flash syntax
+function importFlashMessagesFromRails() {
+  const notifications = $('#notifications').data().flash;
+
+  notifications.forEach(([type, message]) => {
+    // normalize rails flash names
+    if (type === 'danger') {
+      type = 'error';
+    }
+
+    showToast({
```

Comment:
my goal is not to use notify, but to export it to consumers.
since there is different logic to notify then that of import (like sticky if type===success, or danger => error) then that logic is separated between the two functions, which then both call the dispatch method.